src/combinatorics/quiver/basic.lean

```diff
@@ -125,8 +125,75 @@ instance empty_quiver (V : Type u) : quiver.{u} (empty V) := ⟨λ a b, pempty
 
 @[simp] lemma empty_arrow {V : Type u} (a b : empty V) : (a ⟶ b) = pempty := rfl
 
-
 /-- A quiver is thin if it has no parallel arrows. -/
 @[reducible] def is_thin (V : Type u) [quiver V] := ∀ (a b : V), subsingleton (a ⟶ b)
 
+section push
+/-!
+### Pushing the quiver structure on `V` along a map `V → W`
+-/
+
+variables {V : Type*} [quiver V] {W : Type*} (σ : V → W)
+
+/-- The `quiver` instance obtained by pushing arrows of `V` along the map `σ : V → W` -/
+@[nolint unused_arguments]
+def push (σ : V → W) := W
+
+instance [h : nonempty W] : nonempty (push σ) := h
+
+/-- The quiver structure obtained by pushing arrows of `V` along the map `σ : V → W` -/
+@[nolint has_nonempty_instance]
+inductive push_quiver {V : Type u} [quiver.{v} V] {W : Type u₂} (σ : V → W) :
+  W → W → Type (max u u₂ v)
+| arrow {X Y : V} (f : X ⟶ Y) : push_quiver (σ X) (σ Y)
+
+instance : quiver (push σ) := ⟨push_quiver σ⟩
+
+namespace push
+
+/-- The prefunctor induced by pushing arrows via `σ` -/
+def of : V ⥤q push σ :=
+{ obj := σ,
+  map := λ X Y f, push_quiver.arrow f}
+
+@[simp] lemma of_obj : ((of σ)).obj = σ := rfl
+
+variables {W' : Type*} [quiver W'] (φ : prefunctor V W') (τ : W → W') (h : ∀ x, φ.obj x = τ (σ x) )
+
+include φ h
+/-- Any map `φ : V → W'` factoring through `τ : W → W'` lifts to a prefunctor from `V` to `W`. -/
+def lift : push σ ⥤q W' :=
+{ obj := τ,
+  map := @push_quiver.rec V _ W σ
+    (λ X Y f, τ X ⟶ τ Y)
+    (λ X Y f, by { rw [←h X, ←h Y], exact φ.map f, }) }
+
+lemma lift_spec_obj : (lift σ φ τ h).obj = τ := rfl
```

Comment:
Why calling the def `lift` but the lemmas `lift_spec`?

Reply:
That's how I understood it to work: `lift` for the map of the UMP, `lift_spec` for its property, and `lift_unique` for it being unique satisfying this property.

Reply:
Yeah but this is not the "property", this is a projection equality.

Reply:
To me, this, along with `lift_spec_comp` is exactly the property, as witnessed by the fact that in `lift_unique`, exactly those two conditions are required of `\Phi` to prove it is equal to the lift.
What would you rather call that?

Reply:
Simply `lift_obj` and `comp_lift`